design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,148 @@
+- Utilize a familiar and deterministic pattern when injecting bucket connection information into workload environments. 
+
+## Non-Goals
+
+- This proposal does not implement a generalized method for bucket provisioning; it will be specific to Rook-Ceph
+- This design does not attempt to implement a 1:1 model of Kubernetes PersistentVolumes and PersistentVolumeClaims.  Some shallow similarities will exist. 
+- This design does not include the Swift interface implementation by Ceph Object.
+- This design does not provide users a means of deleting buckets via the Kubernetes API.  This creates a dangerous situation which could lead to accidental loss of data.  Only the Kubernetes API objects are deleted.
+
+## Users
+
+- Admin: A Kubernetes cluster administrator with permissions to instantiate and control access to cluster storage. The admin will use existing Rook-Ceph CRDs to create object stores and StorageClasses to make them accessible to users.  
+- User: A Kubernetes cluster user with limited permissions, typically confined CRUD operations on Kubernetes objects within a single namespace.  The user will create request ceph object buckets by instantiating CephObjectBucketClaims. 
+
+## Use Cases
+
+### Assumptions
+
+1. A Rook-Ceph operator deployed on a Kubernetes Cluster.
+1. A CephObjectStore object exists along with the accompanying Service in a protected namespace.
+1. An existing CephObjectStoreUser exists along with the accompanying Secret in the user's namespace.
+
+##### Use Case: Expose an Object Store Endpoint for Bucket Provisioning
+
+As an admin, I want to expose an existing object store to cluster users so they can begin bucket provisioning via the Kubernetes API._
+
+![admin-actions](./cobc-admin.png)
+
+1. The admin creates a StorageClass with `parameters[serviceName]=objectstore-service` and `parameters[serviceNamespace]=some-namespace`
+  - _Note:_ the `provisioner` field is ignored.  Defining the provisioner is unnecessary as Rook-Ceph is the only operator watching for CephObjectBucketClaims.
+
+**Use Case: Provision a Bucket** 
+
+_As a Kubernetes user, I want to leverage the Kubernetes API to create Ceph Object Buckets. I expect to get back the bucket connection information in a Pod-attachable object._
+ 
+![user-actions](./cobc-user.png)
+
+0. The user creates a CephObjectBucketClaim in the user's namespace.
+1. The Rook Operator detects a new ObjectBucketClaim instance.  
+    1. The operator gets the user's credentials via `cephObjectBucketClaim.spec.secretName`.  
+    1. The operator gets the object store URL via `storageClass.parameters.serviceNamespace` + `storageClass.parameters.serviceName`.
+1. The operator uses the object store URL and credentials to make an S3 "make bucket" call.
+1. The operator creates a ConfigMap in the namespace of the CephObjectBucketClaim with relevant connection data for that bucket.
```

Comment:
What happens if the CM already exists?

Reply:
As a pattern, rook typically updates a resource that already exists, but it can depend on the scenario.

Reply:
> What happens if the CM already exists?

If it's a Rook idiom, then we should follow that unless we can think of a reason why we shouldn't.